appinventor/components/src/com/google/appinventor/components/runtime/SingleValueSensor.java

```diff
@@ -0,0 +1,190 @@
+// -*- mode: java; c-basic-offset: 2; -*-
+// Copyright 2009-2011 Google, All Rights reserved
+// Copyright 2019 MIT, All rights reserved
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+
+package com.google.appinventor.components.runtime;
+
+import com.google.appinventor.components.annotations.DesignerProperty;
+import com.google.appinventor.components.annotations.SimpleObject;
+import com.google.appinventor.components.annotations.SimpleProperty;
+import com.google.appinventor.components.common.PropertyTypeConstants;
+
+import android.content.Context;
+import android.hardware.Sensor;
+import android.hardware.SensorEvent;
+import android.hardware.SensorEventListener;
+import android.hardware.SensorManager;
+import android.os.Build;
+
+import java.util.List;
+
+/**
+ * A sensors reporting a single value, such as temperature or humidity.
+ */
+@SimpleObject
+public abstract class SingleValueSensor extends AndroidNonvisibleComponent
+    implements OnStopListener, OnResumeListener, SensorComponent, SensorEventListener, Deleteable {
+  private static final int DEFAULT_REFRESH_TIME = 1000; // ms
+  private Sensor sensor;
+  protected int sensorType;
+  protected float value;  // most recent value read
+  protected final SensorManager sensorManager;
+  protected boolean enabled;
+  protected int refreshTime;
```

Comment:
Since we only send property values if they aren't set to default, you should initialize refreshTime to be DEFAULT_REFRESH_TIME. Otherwise, it will default to 0 and so the refresh rate will be much faster than expected.

Reply:
That would be bad. Thanks very much for catching this.